rpc/wrtc_signaling_server.go

```diff
@@ -152,8 +152,37 @@ func HeartbeatsAllowedFromCtx(ctx context.Context) bool {
 	return md[HeartbeatsAllowedMetadataField][0] == "true"
 }
 
+func (srv *WebRTCSignalingServer) asyncSetOfferError(host string, uuid string, offerErr error) {
+	srv.callMu.RLock()
+	// Atomically check if the cancelCtx was canceled, and if not, add to the `callWorkers`.
+	if err := srv.cancelCtx.Err(); err != nil {
+		srv.callMu.RUnlock()
+		return
+	}
+	srv.callWorkers.Add(1)
+	srv.callMu.RUnlock()
+
+	utils.PanicCapturingGo(func() {
+		defer srv.callWorkers.Done()
+
+		// Use a timeout to not block shutdown.
+		sendCtx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+		defer cancel()
+
+		sendErr := srv.callQueue.SendOfferError(sendCtx, host, uuid, offerErr)
```

Comment:
Also referenced in the deleted code: a "context timeout" that's concurrent with a "return nil // the connection has succeeded" should never result in this `SendOfferError` being invoked.